aci/resource_aci_aaa_domain_relationship.go

```diff
@@ -0,0 +1,188 @@
+package aci
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/ciscoecosystem/aci-go-client/client"
+	"github.com/ciscoecosystem/aci-go-client/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAciDomainRelationship() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAciDomainRelationshipCreate,
+		UpdateContext: resourceAciDomainRelationshipUpdate,
+		ReadContext:   resourceAciDomainRelationshipRead,
+		DeleteContext: resourceAciDomainRelationshipDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceAciDomainRelationshipImport,
+		},
+
+		SchemaVersion: 1,
+		Schema: AppendBaseAttrSchema(AppendNameAliasAttrSchema(map[string]*schema.Schema{
+			"parent_dn": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"aaa_domain_dn": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		})),
+	}
+}
+
+func getRemoteAaaDomainRelationship(client *client.Client, dn string) (*models.AaaDomainRef, error) {
+	aaaDomainRefCont, err := client.Get(dn)
+	if err != nil {
+		return nil, err
+	}
+	aaaDomainRef := models.AaaDomainRefFromContainer(aaaDomainRefCont)
+	if aaaDomainRef.DistinguishedName == "" {
+		return nil, fmt.Errorf("AaaDomainRef %s not found", aaaDomainRef.DistinguishedName)
+	}
+	return aaaDomainRef, nil
+}
+
+func setAaaDomainRelationshipAttributes(aaaDomainRef *models.AaaDomainRef, d *schema.ResourceData) (*schema.ResourceData, error) {
+	dn := d.Id()
+	d.SetId(aaaDomainRef.DistinguishedName)
+	if dn != aaaDomainRef.DistinguishedName {
+		d.Set("parent_dn", "")
+	}
+	aaaDomainRefMap, err := aaaDomainRef.ToMap()
+	if err != nil {
+		return d, err
+	}
+	d.Set("name", GetMOName(aaaDomainRef.DistinguishedName))
```

Comment:
shouldn't this here be setting aaa_domain_dn? Or do you only have name in the state file?